frontend/src/scenes/insights/InsightNav/insightNavLogic.tsx

```diff
@@ -136,6 +136,10 @@ export const insightNavLogic = kea<insightNavLogicType>([
     listeners(({ values, actions }) => ({
         setActiveView: ({ view }) => {
             if (values.isUsingDataExploration) {
+                // setting insight tab is treated as if it is a navigation
+                // but, it's not (always), so, when setting the active view
+                // first clear the existing "new" insight
+                actions.resetNewInsight()
```

Comment:
This is the fix here. When we switch between tabs in the insight editor rather than navigate to it we weren't updating the insightLogic